sympy/tensor/array/array_derivatives.py

```diff
@@ -85,7 +86,7 @@ def _dispatch_eval_derivative_n_times(cls, expr, v, count):
         # object, the default in `Basic` will call a loop over
         # `_eval_derivative`:
 
-        if not isinstance(count, (int, Integer)) or ((count <= 0) == True):
+        if not isinstance(count, (int, Integer, numbers.Integral)) or ((count <= 0) == True):
```

Comment:
No idea if I understand correctly (it's really hard to trace down how you arrive here), but I think this shouldn't be needed if the count in `function.py` is converted.

Reply:
Yes, that's what I thought. I also haven't traced how to get here because the differentiation code is so convoluted now.